ios/RCTPSPDFKit/RCTPSPDFKitView.m

```diff
@@ -142,6 +143,62 @@ - (void)pdfViewController:(PSPDFViewController *)pdfController willBeginDisplayi
   }
 }
 
+#pragma mark - Instant JSON
+
+- (NSDictionary *)getAnnotations:(PSPDFPageIndex)pageIndex type:(PSPDFAnnotationType)type {
+  NSArray <PSPDFAnnotation *>* annotations = [self.pdfController.document annotationsForPageAtIndex:pageIndex type:type];
+  NSArray <NSDictionary *> *annotationsJSON = [RCTConvert instantJSONAnnotationsFromPSPDFAnnotationArray:annotations];
+  return @{@"annotations" : annotationsJSON};
+}
+
+- (void)addAnnotation:(NSString *)jsonAnnotation {
+  if (jsonAnnotation.length == 0) {
+    NSLog(@"Invalid JSON Annotation.");
+    return;
+  }
+
+  NSData *data = [jsonAnnotation dataUsingEncoding:NSUTF8StringEncoding];
+  PSPDFDocument *document = self.pdfController.document;
+  PSPDFDocumentProvider *documentProvider = document.documentProviders.firstObject;
+
+  BOOL success = NO;
+  if (data) {
+    PSPDFAnnotation *annotation = [PSPDFAnnotation annotationFromInstantJSON:data documentProvider:documentProvider error:NULL];
+    success = [document addAnnotations:@[annotation] options:nil];
+  }
+
+  if (!success){
+    NSLog(@"Failed to add annotation.");
+  }
+}
+
+- (NSDictionary *)getAllUnsavedAnnotations {
+  [self.pdfController.document saveWithOptions:nil error:NULL];
+  PSPDFDocumentProvider *documentProvider = self.pdfController.document.documentProviders.firstObject;
+  NSData *data = [self.pdfController.document generateInstantJSONFromDocumentProvider:documentProvider error:NULL];
```

Comment:
Won't this return **all** annotations?
Also, do you think it might be unexpected to save the document in this method, since it looks like a getter from the outside.